### PR TITLE
Reject min channels

### PR DESCRIPTION
**Pull request by niftynei**
I realized after doing this that we have `min-capacity-sat` which lets us reject a channel open who's effective capacity will be below a certain limit.

This is less useful for v2 channels, since we're adding funds to it as well, but you can simply set `min-their-funding` to the same amount and achieve roughly the same thing.

Making this a draft since I don't think we should add it, but open to second opinions.

### CODE REVIEW

**Comment by wtogami** [comment]
> I realized after doing this that we have min-capacity-sat which lets us reject a channel open who's effective capacity will be below a certain limit.

This PR might seem to be redundant given the current effect of  `min-capacity-sat` but that older option is **confusing** from a user point of view in that it isn't an exact number. Options:
* It wouldn't be harmful to instead redefine the meaning of `min-capacity-sat`?
* Could also consider deprecation of the old option. Conceptually `min-their-funding` is clearer in intended meaning for both open1 and open2?

What error message does it tell the other side currently if `min-capacity-sat` is not satisfied?

**Comment by niftynei** [comment]
The error we send to the peer is

```
				      "channel capacity with funding %s,"
				      " reserves %s/%s,"
				      " max_htlc_value_in_flight_msat is %s,"
				      " channel capacity is %s, which is below %s",
```

**Comment by wtogami** [comment]
> The error we send to the peer is
> 
> ```
> 				      "channel capacity with funding %s,"
> 				      " reserves %s/%s,"
> 				      " max_htlc_value_in_flight_msat is %s,"
> 				      " channel capacity is %s, which is below %s",
> ```

That seems a bit too wordy? Some human wants to read it and know how big of a channel they are expected. They don't care about the reserves size or max_htlc_value_in_flight at this stage?
